server/conn.go

```diff
@@ -2341,7 +2341,11 @@ func (cc *clientConn) handleChangeUser(ctx context.Context, data []byte) error {
 	if err := cc.ctx.Close(); err != nil {
 		logutil.Logger(ctx).Debug("close old context failed", zap.Error(err))
 	}
-	cc.ctx = nil
+	// new openSession here, openSessionAndDoAuth will not run again because ctx was not nil.
+	err := cc.openSession()
```

Comment:
Is it ok not to auth?

Reply:
I think so, as you can see `openSessionAndDoAuth` in line 2349.

Reply:
However, is it fine to only re-set the context here? like:

```
	cc.ctx, err = cc.server.driver.OpenCtx(cc.connectionID, cc.capability, cc.collation, cc.dbname, tlsStatePtr)
	if err != nil {
		return err
	}
```

https://github.com/pingcap/tidb/blob/334508e13d556aef1f1f6f766aacc9c70b44626c/server/conn.go/#L798-L801

Reply:
Also, I just checked the reference of `openSessionAndDoAuth()`, it might be fine if we just remove:

```
if cc.ctx == nil {
```
In `openSessionAndDoAuth`? https://github.com/pingcap/tidb/blob/334508e13d556aef1f1f6f766aacc9c70b44626c/server/conn.go/#L812

@dveeden WDYT?

Reply:
sometimes reuse the `ctx` , maybe can not remove 
in function `clientConn::readOptionalSSLRequestAndHandshakeResponse`, will  call function `handleAuthPlugin` new ctx; and then call  `clientConn::openSessionAndDoAuth` will not new ctx (reuse the ctx created before)
reuse code see here : https://github.com/pingcap/tidb/blob/master/server/conn.go#L692-L709